app/controllers/api/v0/api_controller.rb

```diff
@@ -33,9 +33,14 @@ def error_not_found
         render json: { error: "not found", status: 404 }, status: :not_found
       end
 
+      # @note This method is performing both authentication and authorization.  The user suspended
+      #       should be something added to the corresponding pundit policy.
       def authenticate!
-        @user = authenticated_user
-        return error_unauthorized unless @user && !@user.suspended?
+        user = authenticate_with_api_key_or_current_user
+        return error_unauthorized unless user
+        return error_unauthorized if @user.suspended?
```

Comment:
the duplicate error_unauthorized lines here permit changing what/where this check happens (or what error message we give) - currently no behavior change.